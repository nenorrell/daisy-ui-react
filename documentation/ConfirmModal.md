⬅️ [Back to component index](README.md)

# ConfirmModal

The `ConfirmModal` component is a reusable confirm modal component that can be used to create various types of confirm modals, such as delete confirmations and form submission confirmations.

It is built on top of the [ModalFrame Component](ModalFrame.md)

For Modal styling specifics, please see: https://daisyui.com/components/modal/

## Props

| Prop | Type | Required | Default | Description |
| --- | --- | --- | --- | --- |
| `id` | string | optional |  | The id of the modal element. |
| `className` | string | optional |  | The class name to apply on the modal element. |
| `variant` | string | optional | "neutral" | The color variant of the modal. |
| `isOpen` | boolean | required |  | Indicates whether the modal is open or closed. |
| `position` | string | optional | "middle" | The position of the modal. The options are "bottom" and "middle". |
| `closeButtonVariant` | string | optional |  | The variant of the close button. |
| `closeButtonText` | string | optional | "Close" | The text to be displayed on the close button. |
| `confirmButtonVariant` | string | optional |  | The variant of the confirm button. |
| `confirmButtonText` | string | optional | "Confirm" | The text to be displayed on the confirm button. |
| `confirmHandler` | function | required |  | A callback function to handle the confirm event of the modal. |
| `cornerClose` | boolean | optional | false | Indicates whether to show the close button on the top right corner of the modal. |
| `overlayClose` | boolean | optional | true | Indicates whether to close the modal when the overlay is clicked. |


## Examples

```javascript
import { ConfirmModal } from 'daisy-ui-react-components';

function App() {
  const [showModal, setShowModal] = useState(false);
  const handleConfirm = () => {
    // handle confirmation logic
  };

  return (
    <div>
      <button onClick={() => setShowModal(true)}>Show Confirm Modal</button>
      <ConfirmModal 
        isOpen={showModal} 
        closeHandler={() => setShowModal(false)} 
        confirmHandler={handleConfirm}
        confirmButtonText="Delete"
        confirmButtonVariant="danger"
      >
        Are you sure you want to delete this item?
      </ConfirmModal>
    </div>
  );
}
```
