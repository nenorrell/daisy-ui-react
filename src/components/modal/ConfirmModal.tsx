import { forwardRef, MouseEventHandler, PropsWithChildren } from "react";
import { BaselessVariant } from "../../@types/Colors";
import { Button } from "../buttons/Button";
import { IModalFrame, ModalFrame } from "./ModalFrame";

interface IConfirmModal extends IModalFrame{
    confirmButtonVariant ?:BaselessVariant
    confirmHandler ?:MouseEventHandler
    confirmButtonText ?:string
    closeButtonVariant ?:BaselessVariant
    closeButtonText ?:string
}
export const ConfirmModal = forwardRef<HTMLDivElement, PropsWithChildren<IConfirmModal>>((
    {
        closeButtonText,
        closeButtonVariant,
        confirmButtonText,
        confirmButtonVariant,
        confirmHandler,
        children,
        ...props
    },
    ref
)=>{
    return (
        <ModalFrame
            ref={ref}
            {...props}
        >
            {children}

            <div className="modal-action">
                <Button variant={closeButtonVariant} onClick={props.closeHandler}>
                    {closeButtonText || "Close"}
                </Button>

                <Button variant={confirmButtonVariant} onClick={confirmHandler}>
                    {confirmButtonText || "Confirm"}
                </Button>
            </div>
        </ModalFrame>
    );
});

ConfirmModal.displayName = "ConfirmModal";
