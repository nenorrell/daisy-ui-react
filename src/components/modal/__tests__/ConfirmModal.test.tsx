import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { ConfirmModal } from "../ConfirmModal";

describe("ConfirmModal", () => {
    it("should render with default values", () => {
        const { container } = render(<ConfirmModal isOpen={true} closeHandler={jest.fn()} />);
        expect(container).toMatchSnapshot();
    });

    it("should call closeHandler when close button is clicked", () => {
        const closeHandler = jest.fn();
        const { container } = render(<ConfirmModal isOpen={true} closeHandler={closeHandler} />);
        fireEvent.click((container as any).querySelector(".modal-action button:first-child"));
        expect(closeHandler).toHaveBeenCalled();
    });

    it("should call confirmHandler when confirm button is clicked", () => {
        const confirmHandler = jest.fn();
        const { container } = render(<ConfirmModal isOpen={true} closeHandler={jest.fn()} confirmHandler={confirmHandler} />);
        fireEvent.click((container as any).querySelector(".modal-action button:last-child"));
        expect(confirmHandler).toHaveBeenCalled();
    });
});
