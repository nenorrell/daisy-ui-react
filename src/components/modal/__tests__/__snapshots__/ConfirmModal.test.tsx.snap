// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal should render with default values 1`] = `
<div>
  <div
    class="modal modal-open"
  >
    <div
      class="modal-box bg-neutral text-neutral-content"
    >
      <div
        class="modal-action"
      >
        <button
          class="btn btn-neutral btn-md"
        >
          Close
        </button>
        <button
          class="btn btn-neutral btn-md"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
